docs/csharp/whats-new/csharp-7-1.md

```diff
@@ -0,0 +1,187 @@
+following lines:
+
+```xml
+<PropertyGroup>
+  <LangVersion>latest</LangVersion>
+</PropertyGroup>
+```
+
+> [!NOTE]
+> If you use the Visual Studio IDE to update your csproj files, the IDE
+> creates separate nodes for each build configuration. You'll typically
+> set the value the same in all build configurations, but you need to
+> set it explicitly for each build configuration, or select "All Configurations"
+> when you modify this setting. You'll see the following in your csproj file:
+
+```xml
+<PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Release|AnyCPU'">
+  <LangVersion>latest</LangVersion>
+</PropertyGroup>
+
+<PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|AnyCPU'">
+  <LangVersion>latest</LangVersion>
+</PropertyGroup>
+```
+
+Valid settings for the `LangVersion` element are:
+
+* `ISO-1`
+* `ISO-2`
+* `3`
+* `4`
+* `5`
+* `6`
+* `7`
+* `7.1`
+* `default`
+* `latest`
+
+The special strings `default` and `latest` resolve to the latest major
+and minor language versions installed on the build machine, respectively.
+
+This setting decouples installing new versions of the SDK and tools
+in your development environment from choosing to incorporate new language
+features in a project. You can install the latest SDK and tools on your
+build machine. Each project can be configured to use a specific version
+of the language for its build.
+
+## Async main
+
+An *async main* method enables you to use `await` in your `Main` method.
+Previously you would need to write:
+
+```csharp
+static int Main()
+{
+    return DoAsyncWork().GetAwaiter().GetResult();
+}
+```
+
+You can now write:
+
+```csharp
+static async Task<int> Main()
+{
+    // This could also be replaced with the body
+    // DoAsyncWork, including its await expressions:
+    return await DoAsyncWork();
+}
+```
+
+If your program doesn't return an exit code, you can declare a `Main` method
+that returns a <xref:System.Threading.Tasks.Task>:
+
+```csharp
+static async Task Main()
+{
+    await SomeAsyncMethod();
+}
+```
+
+You can read more about the details in the
+[async main](../programming-guid/main-and-command-args/index.md) topic
+in the programming guide.
+
+## Default literal expressions
+
+Default literal expressions are an enhancement to default value expressions.
+These expressions initialize a variable to the default value. Where you previously
+would write:
+
+```csharp
+Func<string, bool> whereClause = default(Func<string, bool>);
+```
+
+You can now omit the type on the right-hand side of the initialization:
+
+```csharp
+Func<string, bool> whereClause = default;
+```
+
+You can learn more about this enhancement in the C# Programming Guide topic
+on [default value expressions](../programming-guide/statements-expressions-operators/default-value-expressions.md).
+
+This enhancement also changes some of the parsing rules for the [`default` keyword](../language-reference/keywords/default.md)
```

Comment:
nit: this link behaves kind of weird because of the inline code in the middle. 
![image](https://user-images.githubusercontent.com/12971179/29696652-2838ea98-88ff-11e7-99d8-37514ab2775d.png)

also this is missing punctuation.